### ISSUE TITLE
Plot an image

### ISSUE THREAD

**Issue by yakir12** [JuliaPlots/Plots.jl#1263]
The original idea was to plot an RGB image.
This lead to these:
```julia
using Plots, Images
img = colorview(RGB, rand(3, 100,100))

# these work:
plot(img)
heatmap(img)

x = 1:100
y = 1:100
# this draws lines, instead of changing the scale of the axis:
plot(x, y, img)
# this changes the colomap, instead of changing the scale of the axis:
heatmap(x, y, img)
```

**Comment by rand5** [comment]
There is a section in the documentation that shows which scales are supported in which backends. They are set using symbols. See [here](http://docs.juliaplots.org/latest/supported/)

**Comment by yakir12** [comment]
I assume you are referring to my comment about the scales of the axis. I'm sorry to have mislead you. I didn't mean scales as in logarithmic or linear. I meant that if I plot an image, the `x` and `y` axis will denote the pixel locations. But if I want to change that to some other specific `x` and `y` (e.g. `3:6:600`) then it seems natural to me that `heatmap(x, y, img)` should work (or instead of `heatmap` use some other dedicated plotting function intended for images). But it doesn't.

**Comment by rand5** [comment]
The range of the axes is set via `xlim` and `ylim`, for example `heatmap(img,xlim=(1,50))` or `plot(img,xlim=(1,50))`. If you want to plot every nth pixel, the most straight forward to do this is to down-sample the input data, i.e. `heatmap(img[1:5:end,1:5:end])`. Axes are addressed [here](http://docs.juliaplots.org/latest/attributes/#axis-and-xaxisyaxiszaxis) in the docs. The behavior of `heatmap(x, y, img)` is the intended behavior.

**Comment by daschw** [comment]
I see what you are expecting @yakir12.
You'd like
```julia
using Plots, Images
img = colorview(RGB, rand(3, 10,10))
plot(10:10:100, 100:100:1000, img)
```
to produce something like this, right?:
![image](https://user-images.githubusercontent.com/16589944/33382083-0fb3523e-d520-11e7-91ba-f1624b11fdaa.png)

I did this with `plot(img, xformatter = x -> 10x, yformatter = y -> 100y)`

That is not possible right now, because in the image recipe the x and y values are fixed to 1:m, 1:n here https://github.com/JuliaPlots/Plots.jl/blob/master/src/series.jl#L328

On the one hand, I see your point that your expected behavior would be natural (it works with heatmap without images: `heatmap(10:10:100, 100:100:1000, rand(10, 10)`) and I don't understand the image that is produced instead:
![image2](https://user-images.githubusercontent.com/16589944/33382356-d1066106-d520-11e7-8766-c05ecede651a.png)

On the other hand, I'm not really sure what such images should look like, if the vectors `x` and `y` you want to pass are not ranges, but arbitrary vectors (like `rand(10)`) instead. After all, `x` and `y` denote the the discrete positions of the pixels to be colored in a matrix. So maybe using `formatter` or passing `ticks = (locations, labels)` is the cleaner solution?

**Comment by Evizero** [comment]
> `img = colorview(RGB, rand(3, 10,10))`

FWIW you don't need to use this 2-step process to generate random images.

```julia
julia> img = rand(RGB, 10, 10)
10×10 Array{RGB{Float64},2}:
[...]
```

**Comment by yakir12** [comment]
You are correct @daschw. Thank you!
The x and y should only be ranges, no need for random locations (not sure what that would mean either). So your `xformatter = x -> 10x` solution is cool! So for an offset I'd just do `xformatter = x -> 10x + x_offset`.

**Comment by yakir12** [comment]
So I tried it a bit and the problem now is the number of digits after the decimal point when the formater includes fractional numbers. So something like this:
```julia
plot(img, xformatter = x -> rand()*x, yformatter = y -> 3*rand()*y)
```
looks like this:
![tmp](https://user-images.githubusercontent.com/3281957/33519352-570ac61c-d7a5-11e7-834c-c64c0a720bd1.png)

**Comment by daschw** [comment]
You could just use `round` in the formatter:
```julia
plot(img, xformatter = x -> round(rand()*x, 2), yformatter = y -> round(3*rand()*y, 2))
```

**Comment by yakir12** [comment]
🤦‍

**Comment by natschil** [comment]
@rand5: the approach you suggests seems to currently be broken if one wants to the y-axis to go from negative below to positive above, e.g. doing something like ylim=(90.0,-90.0) on a world-map to set the correct Longitudes gives me ylim scaled from 90.00 (on top) to 91.00 (on the bottom), i.e. the direction is still wrong...

**Comment by BA1437** [comment]
@natschil Could you post a minimum working example?

**Comment by natschil** [comment]
@BA1437 The image I'm currently using is a bit large, but if you have some image file the following fails for me:

`using FileIO
image = load("path_to_image.tif")
plot(image, ylim=(90.0,-90.0))
`

The y-coordinate is always increasing when going down.

**Comment by BA1437** [comment]
See https://github.com/JuliaPlots/Plots.jl/issues/1196

**Comment by daschw** [comment]
`ylim` always is a tuple `(ymin, ymax)` with `ymin < ymax`. If `ymin >= ymax`, Plots sets `ymax = ymin + 1`. The `ymin` value just appears above the `ymax` value for images because `yflip` is set `true` in that case.
So in your case @natschil `ylim = (-90, 90)` should do what you expect, right?

**Comment by daschw** [comment]
@natschil If you want the image to be flipped vertically you can do `yflip = false`, If you only want the tick labels to be changed from negative to positive and vice versa you can do `yformatter = x -> -x`

**Comment by natschil** [comment]
@daschw I don't want the image to be flipped vertically, but the coordinates used by Plots.jl (i.e. I want scatter!(xs,ys) to correctly show up on top of the image. If yflip = false is supposed to do this then there is a bug somewhere (it has absolutely no affect as far as I can tell on GR)

**Comment by daschw** [comment]
```julia
using Plots
using TestImages
img = testimage("lighthouse")

# plot the image reversing the first dimension and setting yflip = false
plot([-π, π], [-1, 1], img[end:-1:1, :], yflip = false)
# plot other data
plot!(sin, -π, π, lw = 3, color = :red)
```

![lighthouse_sin](https://user-images.githubusercontent.com/16589944/37529794-4f172d62-2938-11e8-93d1-611e0814cf33.png)

**Comment by natschil** [comment]
This is broken in 1.0, running the code above results in the following image:
![output](https://user-images.githubusercontent.com/3118879/44684853-6a265900-aa4a-11e8-81d4-7faa2fa1b4af.png)

**Comment by mkborregaard** [comment]
Wow, thanks for reporting. So, the `plot(img[end:-1:1,:])` actually works in itself, it's adding the axes that makes it fail

**Comment by natschil** [comment]
Yes. Do you know what could be causing this/whether there is a workaround?

**Comment by logankilpatrick** [comment]
Hello, any update on this? It worked for me a few months ago, but now it’s not working again.

**Comment by mkborregaard** [comment]
no updates, sorry.

**Comment by logankilpatrick** [comment]
It used to work for me in Julia v0.7, without selecting a backend. Now it does not work. Any work arounds?

**Comment by mkborregaard** [comment]
Could you try the plotly backend? That used to be the default.

**Comment by daschw** [comment]
Should be fixed in #1951.
